File: backend/src/main/java/com/back/domain/item/item/dto/ItemCreateResponse.java
package com.back.domain.item.item.dto;

import com.back.domain.item.item.entity.Item;
import com.back.domain.item.item.util.DDayCalculator;

import java.time.LocalDate;

public record ItemCreateResponse(
        Long id,
        Long userId,
        Long categoryId,
        String categoryName,
        String name,
        String imgUrl,
        LocalDate startDate,
        String cycleDays,
        LocalDate nextReplacementDate,
        Long dDay,
        Boolean isActive
) {
    /**
     * 생성된 Item Entity를 Response DTO로 변환
     *
     * @param item 생성된 Item 엔티티
     * @return ItemCreateResponse DTO
     */
    public static ItemCreateResponse from(Item item) {
        return new ItemCreateResponse(
                item.getId(),
                item.getUser().getId(),
                item.getCategory() == null ? null : item.getCategory().getId(),
                item.getCategory() == null ? null : item.getCategory().getName(),
                item.getName(),
                item.getImgUrl(),
                item.getStartDate(),
                item.getCycleDays(),
                item.getNextReplacementDate(),
                DDayCalculator.calculate(item.getNextReplacementDate()),
                item.getIsActive()
        );
    }
}

File: backend/src/main/java/com/back/domain/item/item/dto/ItemCreateResponse.kt
package com.back.domain.item.item.dto

import com.back.domain.item.item.entity.Item
import com.back.domain.item.item.util.DDayCalculator
import java.time.LocalDate

@JvmRecord
data class ItemCreateResponse(
    val id: Long?,
    val userId: Long?,
    val categoryId: Long?,
    val categoryName: String?,
    val name: String?,
    val imgUrl: String?,
    val startDate: LocalDate?,
    val cycleDays: String?,
    val nextReplacementDate: LocalDate?,
    val dDay: Long?,
    val isActive: Boolean?
) {
    companion object {
        /**
         * 생성된 Item Entity를 Response DTO로 변환
         *
         * @param item 생성된 Item 엔티티
         * @return ItemCreateResponse DTO
         */
        @JvmStatic
        fun from(item: Item): ItemCreateResponse {
            return ItemCreateResponse(
                id = item.id,
                userId = item.user?.id,
                categoryId = item.category?.id,
                categoryName = item.category?.name,
                name = item.name,
                imgUrl = item.imgUrl,
                startDate = item.startDate,
                cycleDays = item.cycleDays,
                nextReplacementDate = item.nextReplacementDate,
                dDay = DDayCalculator.calculate(item.nextReplacementDate),
                isActive = item.isActive
            )
        }
    }
}